source/include/agent_message.h

```diff
@@ -33,13 +33,13 @@
 
 /* Declare here so interface functions can use. */
 struct Command;
+struct AgentMessageContext;
```

Comment:
Why was this forward declaration moved above?

Reply:
If the goal is the group the forward declarations, should there be a `@brief` for the `Command_t` type?

Reply:
Yes, it was to group them. Command_t doesn't have a `@brief` here since it has one where the struct is defined in `mqtt_agent.h`. However, it causes warnings in doxygen to define the type and the struct in different files that I haven't figured out yet. One solution could be to move the typedef to the other file, and change everything here to `struct Command`. I left that for the next PR

Reply:
I think that from a reader's perpsective, it is better to be consistent about use of `Command_t` everywhere instead of using `Command_t` in one header file and `struct Command` in another API file. 
Currently, I see that the `mqtt_agent.h` file depends on `agent_message.h` to receive the `Command_t` typedef which can also be confusing to the reader because `struct Command` is defined in `mqtt_agent.h` but the file uses its typedef in the API that it sources from `agent_message.h` file, which may not be obvious.

One solution can be to define a `mqtt_agent_types.h` file where all the types are housed instead of having the structs and their typedefs spread in different files. It can be considered for a different PR though